### PR TITLE
memoizing rlp encoding of serialized nodes

### PR DESCRIPTION
**Pull request by SoujanyaPonnapalli**
This branch implements getBulk by performing a radix-sort on the list of keys. Extends the getBulk-radix-sort branch.

Benchmarks after adding memoization for rlpEncoding of serialized nodes:

```
no-op: 177625±2.85% ops/s 0.006±0.0006 ms/op (57 runs)
put (empty tree): 127820±3.99% ops/s 0.008±0.0014 ms/op (73 runs)
get (empty tree): 119620±5.00% ops/s 0.008±0.0019 ms/op (76 runs)
individual bulk get: 812±29.39% ops/s 1.231±0.6534 ms/op (15 runs)
bulk get: 14176±20.62% ops/s 0.071±0.0229 ms/op (12 runs)
individual get 1: 13640±29.33% ops/s 0.073±0.0372 ms/op (14 runs)
bulk get 1: 82599±43.75% ops/s 0.012±0.0092 ms/op (14 runs)
generate 1k-10k-32-ran: 408±14.15% ops/s 2.450±1.2877 ms/op (53 runs)
generate 1k-1k-32-ran: 43.12±13.35% ops/s 23.192±11.5044 ms/op (53 runs)
generate 1k-1k-32-mir: 45.16±9.62% ops/s 22.145±8.6315 ms/op (63 runs)
generate 1k-9-32-ran: 18.87±6.43% ops/s 53.001±12.0400 ms/op (48 runs)
generate 1k-5-32-ran: 16.55±10.35% ops/s 60.419±20.9188 ms/op (43 runs)
generate 1k-3-32-ran: 14.95±10.99% ops/s 66.899±26.2563 ms/op (49 runs)
```

### CODE REVIEW

**Comment by SoujanyaPonnapalli** [comment]
On my remote branch, I tried using the memoized rlp encodings for the individual get functions and here are the benchmarks:
```
no-op: 199028±1.00% ops/s 0.005±0.0002 ms/op (82 runs)
put (empty tree): 146875±1.20% ops/s 0.007±0.0004 ms/op (80 runs)
get (empty tree): 165540±1.46% ops/s 0.006±0.0004 ms/op (76 runs)
individual bulk get: 14566±50.69% ops/s 0.069±0.0548 ms/op (12 runs)
bulk get: 16861±5.59% ops/s 0.059±0.0055 ms/op (13 runs)
individual get 1: 83846±39.41% ops/s 0.012±0.0078 ms/op (13 runs)
bulk get 1: 117342±3.54% ops/s 0.009±0.0005 ms/op (14 runs)
generate 1k-10k-32-ran: 425±11.85% ops/s 2.351±1.0640 ms/op (56 runs)
generate 1k-1k-32-ran: 50.99±10.30% ops/s 19.612±8.0478 ms/op (61 runs)
generate 1k-1k-32-mir: 44.05±12.14% ops/s 22.702±12.0145 ms/op (73 runs)
generate 1k-9-32-ran: 18.72±8.75% ops/s 53.409±16.5270 ms/op (48 runs)
generate 1k-5-32-ran: 17.20±8.04% ops/s 58.126±17.0273 ms/op (51 runs)
generate 1k-3-32-ran: 15.89±7.81% ops/s 62.942±21.5717 ms/op (74 runs)
```

Now the real question is, should we just add memoization for the rlp encoding of nodes and not memoize node serializations? Also do we want to have bulk gets?

File: src/index.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import {options} from 'benchmark';
 import {hashAsBuffer, HashType} from 'bigint-hash';
 import {RlpDecode, RlpEncode, RlpItem, RlpList} from 'rlp-stream';
+import { sort } from 'shelljs';
```
Comment (no2chem) [inline src/index.ts]: Don't need these...

**Comment by no2chem** [comment]
@SoujanyaPonnapalli Also some small conflicts before we can merge

**Comment by SoujanyaPonnapalli** [comment]
> @SoujanyaPonnapalli Also some small conflicts before we can merge

Yeah the conflict is with the memoization of node serializations that we pushed yesterday. So here is the question:
Do we want to memoize both node serializations and RlpEncoding( serialized nodes) ?
(I'll resolve the conflicts accordingly)

**Comment by no2chem** [comment]
@SoujanyaPonnapalli I think we just want the memoize the RlpItems (this branch) and not the resulting serialized buffers (previous) because it has reduced overhead. 

Memozing the RLPItems only involves keeping a bunch of pointers around, whereas memoizing the RLP encoded items involves plenty of allocations on the V8 stack.

**Comment by SoujanyaPonnapalli** [comment]
Okay then, ill revert the memoization of node serializations and add memoization for the rlpItems.

**Pull request closed without merging**